lib/ansible/modules/cloud/vmware/vmware_export_ovf.py

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Diane Wang <dianew@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vmware_export_ovf
+short_description: Exports a VMware virtual machine to an OVF file, device files and a manifest file
+description: >
+   This module can be used to export a VMware virtual machine to OVF template from vCenter server or ESXi host.
+version_added: '2.8'
+author:
+- Diane Wang (@Tomorrow9) <dianew@vmware.com>
+requirements:
+- python >= 2.6
+- PyVmomi
+notes: []
+options:
+  name:
+    description:
+    - Name of the virtual machine to export.
+    - This is a required parameter, if parameter C(uuid) is not supplied.
+  uuid:
+    description:
+    - Uuid of the virtual machine to export.
+    - This is a required parameter, if parameter C(name) is not supplied.
+  datacenter:
+    default: ha-datacenter
+    description:
+    - Datacenter name of the virtual machine to export.
+    - This parameter is case sensitive.
+  folder:
+    description:
+    - Destination folder, absolute path to find the specified guest.
+    - The folder should include the datacenter. ESX's datacenter is ha-datacenter.
+    - This parameter is case sensitive.
+    - 'If multiple machines are found with same name, this parameter is used to identify
+       uniqueness of the virtual machine. version_added 2.5'
+    - 'Examples:'
+    - '   folder: /ha-datacenter/vm'
+    - '   folder: ha-datacenter/vm'
+    - '   folder: /datacenter1/vm'
+    - '   folder: datacenter1/vm'
+    - '   folder: /datacenter1/vm/folder1'
+    - '   folder: datacenter1/vm/folder1'
+    - '   folder: /folder1/datacenter1/vm'
+    - '   folder: folder1/datacenter1/vm'
+    - '   folder: /folder1/datacenter1/vm/folder2'
+  export_dir:
+    description:
+    - Absolute path to place the exported files on the server running this task, must have write permission.
```

Comment:
I misinterpreted this as "on the ESXi host running this task" and got confused. Perhaps say "on the server running this task" since host is a bit of an overloaded term for VMware.

Reply:
resolved.